grails-app/services/org/pih/warehouse/core/LocationService.groovy

```diff
@@ -41,6 +47,9 @@ class LocationService {
         String name = getReceivingLocationName(shipmentNumber)
         String[] receivingLocationNames = [name, "Receiving ${shipmentNumber}"]
         Location location = findInternalLocation(parentLocation, receivingLocationNames)
+        if (!location) {
+            location = findByLocationNumber(locationNumber)
```

Comment:
The underlying issue is that the receiving location is created when we go to the partial receiving page when logged into one location. 
https://github.com/openboxes/openboxes/blob/develop/grails-app/controllers/org/pih/warehouse/partialReceiving/PartialReceivingController.groovy#L16

And then the user changes the destination and then goes back to the partial receiving page.

It appears as though this doesn't address the issue as it appear that we are changing the parent location of this receiving location. We're just finding it in a different way and then receiving it into another facilities bin location.

Reply:
In addition, this scares me because I assume let's say something created a parent location using the same naming convention that we're using for receiving location. I don't think the receiving and putaway feature would be happy if instead of a bin location, it was given a facility location.

There's also a case where we partially receive into the receiving location in one facility and then change the destination (can we even do that?) and then receive the rest of the items into another facility. 

Let's ask Kelsey how she expects this to behave.